### PR TITLE
Various fixes trying to make it work with our repositories

### PR DESCRIPTION
**Pull request by dmacvicar**
They are in individual commits, but I can highlight:

* Configuration defaults are included, config.yml is overriden and colors.yml is merged.
* Use org as key in repositories as we have the same repo name under two different orgs.
* Initial rubocop file and comma cleanup

### CODE REVIEW

File: db/database.rb [hunk 1]
```diff
@@ -68,9 +68,9 @@ class Organization
 
 class Repository
   include DataMapper::Resource
-  property :rname, String, :required => true, :key => true
+  property :rname, String, :required => true, :key => true, :unique_index => false
 
-  belongs_to :organization, :required => false
+  belongs_to :organization, :required => false, :key => true
```
Comment (MaximilianMeister) [inline db/database.rb]: I fully agree to add this here. The only problem now is that I cannot migrate my existing DB to this. I will have to take a look how a migration could be done here.

```
`execute_non_query': Cannot add a NOT NULL column with default value NULL (DataObjects::SyntaxError)
```

File: lib/nailed.rb [hunk 2]
```diff
@@ -2,19 +2,28 @@
 require 'yaml'
 require "octokit"
 require "bicho"
+require "confstruct"
```
Comment (MaximilianMeister) [inline lib/nailed.rb]: cool, haven't known that gem yet

File: bin/app [hunk 3]
```diff
@@ -356,9 +359,9 @@ class App < Sinatra::Base
   #
   get "/json/github/donut/allpulls" do
     pulltop = []
-    allpulls = Pullrequest.aggregate(:repository_rname, :all.count)
+    allpulls = Pullrequest.aggregate(:repository_organization_oname, :repository_rname, :all.count)
     allpulls.each do |donut_bite|
-      pulltop << {label: donut_bite[0], value: donut_bite[1]}
+      pulltop << {label: "#{donut_bite[0]}/#{donut_bite[1]}", value: donut_bite[2]}
```
Comment (MaximilianMeister) [inline bin/app]: the string could get too long in the donut hole, and subsequently the font size gets too small. but that is a problem of the `morris.js` library which could lead to some ugly svg hacking.

would you be fine with not adding the org_name in front of every repo here in this view?

so just
``` ruby
pulltop << {label: "#{donut_bite[1]}", value: donut_bite[2]}
```

**Comment by MaximilianMeister** [comment]
@dmacvicar There are some double commits in this PR, I think a merge commit slipped in here

**Comment by dmacvicar** [comment]
I will redo the PR, indeed the branch is screwed.

File: bin/app (hunk 3 above)
Comment (dmacvicar) [inline bin/app]: I can fix that.

**Comment by dmacvicar** [comment]
Fixed double commits

File: db/database.rb (hunk 1 above)
Comment (dmacvicar) [inline db/database.rb]: Do you know any way that we could fix the migration in this case?. This fix is important for me as we do have two repos named the same under different orgs.

**Comment by MaximilianMeister** [comment]
@dmacvicar sorry for taking so long but after the ping from @kkaempf I will merge this. can you please rebase?

File: db/database.rb (hunk 1 above)
Comment (MaximilianMeister) [inline db/database.rb]: @dmacvicar you can drop the `:required => false` here

**Comment by kkaempf** [comment]
@MaximilianMeister just close this PR for good, I'll file a new one.

**Pull request closed without merging**